lib/sim.c

```diff
@@ -1181,7 +1247,102 @@ sim_simulate(sim_t *self, uint64_t max_events)
     return ret;
 }
 
+int
+sim_simulate_arg(sim_t *self) 
+{
+    int ret = 0;
+    double recomb_rate, coal_rate, total_rate; 
+    individual_t **S = self->intersected_buffer;
+    unsigned int S_size = 0;
+    avl_node_t *node;
+    uintptr_t int_ptr;
+    uint64_t id;
+    individual_t *ind;
+    /* An arbitrary radius and parental location for using existing spatial 
+     * functions to complete events */
+    double z[] = {self->pixel_size / 2.0, self->pixel_size / 2.0}, r = self->pixel_size / 4.0;
```

Comment:
I'd prefer to keep initialisations and declarations seperate. For this sort of thing I think it's too easy to miss an initialisation or set it to the wrong value if they're mixed on the same line. I would just split these onto two lines.

Also, are these inits supported in c99? I don't really want to use c11 features yet, as it makes portability a pain.